### ISSUE TITLE
Release mono-repo requiring sub-package released version

### ISSUE THREAD

**Issue by davidbrochart** [jupyter-server/jupyter_releaser#371]
### Problem

[Jupyverse](https://github.com/jupyter-server/jupyverse) is a Python mono-repo consisting of a "main" `jupyverse`  package and a bunch of "sub-packages" in [plugins](https://github.com/jupyter-server/jupyverse/tree/main/plugins) (`fps-contents`, `fps-kernels`, etc.).
When releasing e.g. `0.1.0`, I would like `jupyverse` to depend on sub-packages with version `==0.1.0` too. But I cannot pin them in [setup.cfg](https://github.com/jupyter-server/jupyverse/blob/7577605fee6687ef74c6eba69574421dcf75781d/setup.cfg#L24-L31), because at the time of release these sub-packages would not have been released yet.
Maybe I'm missing something?

**Comment by blink1073** [comment]
Right now we are releasing the files in alphabetical order.  We could inspect the dist files' dependencies and release them in dependent order.

**Comment by davidbrochart** [comment]
Yes, but I would still need to make a commit with the sub-package pinning that wouldn't pass the tests, because they would not have been released yet.

**Comment by blink1073** [comment]
Right, for the commit that updates the versions.  I don't see a way around that unless you make a local file that has the next version mapping and the have the version bump use that during release.

**Comment by davidbrochart** [comment]
That would work. The release order could also come from this file.

**Comment by blink1073** [comment]
Interesting idea, a release manifest file.

**Comment by blink1073** [comment]
Oooh, I like the idea of using that for the order.  I'd suggest upgrading to pyproject.toml and adding a general capability to update versions using a manifest file.   You can use `hatch new --init` to automatically convert to [hatch](https://hatch.pypa.io/dev/intro/).

**Comment by blink1073** [comment]
If you want to stick with setuptools, you can use https://github.com/abravalheri/ini2toml

**Comment by davidbrochart** [comment]
Thanks, I'll look at that next week, have a great week-end Steve!

**Comment by blink1073** [comment]
You too!
